x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryRewriter.java

```diff
@@ -35,96 +37,182 @@
-                mustQueryBuilder.must(new TermsQueryBuilder("_id", excludedDoc.id()));
+        protected S getSettings(RelevanceMatchQueryBuilder relevanceMatchQuery, SearchExecutionContext context) {
+            String settingsId = getSettingsId(relevanceMatchQuery);
+            SearchEngine searchEngine = getSearchEngine(context);
 
-                excludedDocsBuilder.mustNot(mustQueryBuilder);
+            if (settingsId == null && searchEngine != null) {
+                settingsId = getSettingsId(searchEngine);
             }
-            booleanQueryBuilder.filter(excludedDocsBuilder);
 
-            queryBuilder = booleanQueryBuilder;
+            if (settingsId != null) {
+                try {
+                    return settingsService.getSettings(settingsId);
+                } catch (SettingsServiceException e) {
+                    throw new IllegalArgumentException("[relevance_match] " + e.getMessage());
+                }
+            }
+
+            return null;
         }
-        return queryBuilder;
-    }
 
-    private static QueryBuilder applyPinnedDocs(QueryBuilder queryBuilder, CurationSettings curationSettings) {
-        if (curationSettings.pinnedDocs().isEmpty() == false) {
-            final PinnedQueryBuilder.Item[] items = curationSettings.pinnedDocs()
-                .stream()
-                .map(docRef -> new PinnedQueryBuilder.Item(docRef.index(), docRef.id()))
-                .toArray(PinnedQueryBuilder.Item[]::new);
-            queryBuilder = new PinnedQueryBuilder(queryBuilder, items);
+        protected abstract String getSettingsId(SearchEngine searchEngine);
+
+        protected abstract String getSettingsId(RelevanceMatchQueryBuilder relevanceMatchQuery);
+
+        private SearchEngine getSearchEngine(SearchExecutionContext context) {
+            if (context.getSearchEngineName() == null) {
+                return null;
+            }
+
+            assert clusterService.state().metadata().searchEngines().containsKey(context.getSearchEngineName());
+
+            return clusterService.state().metadata().searchEngines().get(context.getSearchEngineName());
         }
-        return queryBuilder;
     }
 
-    private Map<String, Float> retrieveFieldsAndBoosts(String relevanceSettingsId, SearchExecutionContext context) {
-        Map<String, Float> fieldsAndBoosts;
-        if (relevanceSettingsId != null) {
-            try {
-                RelevanceSettings relevanceSettings = relevanceSettingsService.getSettings(relevanceSettingsId);
-                QueryConfiguration queryConfiguration = relevanceSettings.getQueryConfiguration();
-                fieldsAndBoosts = queryConfiguration.getFieldsAndBoosts();
-            } catch (AbstractSettingsService.SettingsServiceException e) {
-                throw new IllegalArgumentException("[relevance_match] " + e.getMessage());
+    private static class OrganicQueryRewriter extends AbstractQueryRewriter<RelevanceSettings> {
```

Comment:
Why did we choose the name `OrganicQueryRewriter`? Is this performing the function of a `RelevanceQueryRewriter`?